armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -235,25 +239,55 @@ def _isTopDummyBlockPresent(self):
 
     def axiallyExpandAssembly(self):
         """Utilizes assembly linkage to do axial expansion."""
+        # Set up conservation report information (useful for debugging)
+        self.massConservationReport = {}
```

Comment:
It appears that a lot of work is done to build the `massConservationReport` and `detailedMassConservation` dicts.

I am concerned that this work will be performed on every call to `axiallyExpandAssembly()`, regardless of what the `runLog` verbosity is set to for the run.

It seems like there should be an elegant way to cut out all this work if the verbosity setting is low enough to not be printing these reports.

Reply:
Yeah, I agree. Working on this.